bcda/cclf/cclf.go

```diff
@@ -509,87 +518,96 @@ func validate(fileMetadata *cclfFileMetadata, cclfFileValidator map[string]cclfF
 
 	r, err := zip.OpenReader(filepath.Clean(fileMetadata.filePath))
 	if err != nil {
-		fmt.Printf("Could not read archive %s.\n", fileMetadata)
-		err := errors.Wrapf(err, "could not read archive %s", fileMetadata)
+		fmt.Printf("Could not read archive %s.\n", fileMetadata.filePath)
+		err := errors.Wrapf(err, "could not read archive %s", fileMetadata.filePath)
 		log.Error(err)
 		return err
 	}
 	defer r.Close()
 
 	count := 0
 	validator := cclfFileValidator[key]
+	var rawFile *zip.File
 
-	for i, f := range r.File {
-		fmt.Printf("Reading file #%d from archive %s.\n", i, fileMetadata)
-		log.Infof("Reading file #%d from archive %s", i, fileMetadata)
-		if err := validateFileName(f.Name); err != nil {
-			fmt.Printf("Unknown file name when validating file: %s.\n", f.Name)
-			err = errors.Wrapf(err, "unknown file name when validating file: %s.\n", f.Name)
-			log.Error(err)
-			return err
-		}
-		if err = parseTimestamp(fileMetadata, f.Name); err != nil {
-			return err
-		}
-		rc, err := f.Open()
-		if err != nil {
-			fmt.Printf("Could not read file %s in archive %s.\n", f.Name, fileMetadata)
-			err = errors.Wrapf(err, "could not read file %s in archive %s", f.Name, fileMetadata)
-			log.Error(err)
-			return err
+	for _, f := range r.File {
+		if f.Name == fileMetadata.name {
+			rawFile = f
+			fmt.Printf("Reading file %s from archive %s.\n", fileMetadata.name, fileMetadata.filePath)
+			log.Infof("Reading file %s from archive %s", fileMetadata.name, fileMetadata.filePath)
 		}
-		defer rc.Close()
-		sc := bufio.NewScanner(rc)
-		for sc.Scan() {
-			b := sc.Bytes()
-			bytelength := len(bytes.TrimSpace(b))
-			if bytelength > 0 && bytelength <= validator.maxRecordLength {
-				count++
-
-				// currently only errors if there are more records than we expect.
-				if count > validator.totalRecordCount {
-					fmt.Printf("Maximum record count reached for file %s, Expected record count: %d, Actual record count: %d.\n", key, validator.totalRecordCount, count)
-					err := fmt.Errorf("maximum record count reached for file %s (expected: %d, actual: %d)", key, validator.totalRecordCount, count)
-					log.Error(err)
-					return err
-				}
-			} else {
-				fmt.Printf("Incorrect record length for file %s, Expected record length: %d, Actual record length: %d.\n", key, validator.maxRecordLength, bytelength)
-				err := fmt.Errorf("incorrect record length for file %s (expected: %d, actual: %d)", key, validator.maxRecordLength, bytelength)
+	}
+
+	rc, err := rawFile.Open()
```

Comment:
It is unclear to me what is happening here.  Which file is this opening?  The top-level zip or the text file within the zip?

Reply:
Opening up the file within the zip